### PR TITLE
interop-testing: support rpc keep-open for xDS test server

### PR DESCRIPTION
**Pull request by voidzcy**
Sending out server changes separately to unblock other languages' test client.

/cc @donnadionne @menghanl

### CODE REVIEW

File: interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java [hunk 1]
```diff
@@ -203,18 +214,22 @@ public void setNotServing(
...
+      this.keepOpenCaptor = keepOpenCaptor;
     }
 
     @Override
     public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
         ServerCall<ReqT, RespT> call,
         final Metadata requestHeaders,
         ServerCallHandler<ReqT, RespT> next) {
+      String callBehavior = requestHeaders.get(CALL_BEHAVIOR_KEY);
+      keepOpenCaptor.set("keep-open".equals(callBehavior));
```
Comment (ericgribkoff) [inline interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java:232]: I don't see how the keepOpen atomic's value is correlated with this call after this interceptor sets it to true.

My reading of the spec for these tests was that, for an incoming call with `keep-open` in the metadata, _that_ call should be kept open by the server. As I understand the code here, instead _any_ call with `keep-open` in the metadata essentially flips a global boolean and all calls from that point on will be kept open (at least until another RPC comes without `keep-open`). Is that intended?

To set a value that would instead be scoped to this particular call, I think you could set a value in the context, as in https://github.com/GoogleCloudPlatform/traffic-director-grpc-examples/blob/master/java/src/main/java/io/grpc/examples/wallet/WalletInterceptors.java#L106 (there might be a better way to accomplish this)

Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java:232]: Oops, you are right. This is screwed up, I was too focused on how the test case will be running, in which all calls will be open after some point of time.

Comment (voidzcy) [reply interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java]: Should be fixed now.

**Comment by voidzcy** [comment]
Adding @dapengzhang0 for review as Eric. G is OOO.

**Comment by voidzcy** [comment]
@ericgribkoff Ping.

**Review by ericgribkoff — APPROVED** [review]
(meta-nit: s/interop-testsing/interop-testing/ in the PR title)

**Pull request merged**